pennylane_sf/remote.py

```diff
@@ -159,17 +161,15 @@ def var(self, observable, wires, par):
         return samples_variance(self.samples)
 
     def probability(self, wires=None):  # pylint: disable=missing-function-docstring
-        all_probs = all_fock_probs_pnr(self.samples)
-
-        # Extract the cutoff value by checking the number of Fock states we
-        # obtained probabilities for
-        cutoff = all_probs.shape[0]
+        fock_probs = all_fock_probs_pnr(self.samples)
+        cutoff = fock_probs.shape[0]
+        diff = max(self.cutoff - cutoff, 0)
+        all_probs = np.pad(fock_probs, [(0, diff)] * self.num_wires)
 
         if wires is None:
-
             all_probs = all_probs.flat
             N = self.num_wires
-            ind = np.indices([cutoff] * N).reshape(N, -1).T
+            ind = np.indices([self.cutoff] * N).reshape(N, -1).T
```

Comment:
What happens in the `cutoff > self.cutoff` case here? :thinking: Would some values be disregarded?

Reply:
Yes, we'd be ignoring some values. However, I think this is ok and in line with what one would expect when calculating probabilities with a fixed cutoff, i.e., sometimes you lose some data.

Although I do agree that in this case it could be a shame to throw away useful samples with a high number of photons when it comes from actual hardware. If the user really wants to keep all information, maybe they'd be better to stick with returning samples?

Reply:
There might be subtle differences here between simulation and hardware (and not sure I fully understand the implications).

In the simulation case, we specify a truncated space where all computation takes place. For the Gaussian PNR samples, no information is discarded, but rather the samples might not be an accurate representation since we are always enforcing a probability of 1 across the truncated subspace.

In the hardware case, we are instead discarding additional info?

Reply:
:thinking: yes, in the hardware case we can have samples with any number of photons without a cutoff. We then create `fock_probs`, which necessarily sums to 1 since it includes all of the probabilities. This is then shrunk down if the cutoff is below the cutoff inferred from `fock_probs`.

This is probably different to (a) samples from a simulator with a fixed cutoff and (b) calculating the exact probability distribution. Not really sure what the solution is though other than just being aware.